saas/control_plane/tofu/main.tf

```diff
@@ -0,0 +1,113 @@
+terraform {
+  required_providers {
+    google = {
+      source  = "hashicorp/google"
+      version = "6.14.1"
+    }
+  }
+
+  backend "gcs" {
+    bucket = var.tf_state_bucket_name
+  }
+}
+
+provider "google" {
+  project = var.project_id
+  region  = var.region
+}
+
+locals {
+  services_ip_range_name = "services-range"
+  pods_ip_range_name     = "pods-range"
+}
+
+resource "google_compute_network" "civiform" {
+  name                     = var.network_name
+  description              = "VPC network for the application."
+  enable_ula_internal_ipv6 = true
+  auto_create_subnetworks  = false
+}
+
+resource "google_compute_subnetwork" "civiform" {
+  name             = var.subnetwork_name
+  network          = google_compute_network.civiform.id
+  ip_cidr_range    = "10.0.0.0/16"
+  region           = var.region
+  stack_type       = "IPV4_IPV6"
+  ipv6_access_type = "INTERNAL"
+}
+
+resource "google_container_cluster" "civiform" {
+  name                     = var.cluster_name
+  location                 = var.cluster_location
+  deletion_protection      = var.cluster_deletion_protection
+  network                  = google_compute_network.civiform.id
+  subnetwork               = google_compute_subnetwork.civiform.id
+  datapath_provider        = "ADVANCED_DATAPATH"
+  enable_l4_ilb_subsetting = true
+
+  # We can't create a cluster with no node pool defined, but we want to only use
+  # separately managed node pools. So we create the smallest possible default
+  # node pool and immediately delete it.
```

Comment:
Is this a GCP or TF thing?

Reply:
Kinda both. GKE creates new clusters with a default node pool, and that can't be disabled. The option to immediately delete it is implemented (with hard-coded nodepool name 😬) in TF: https://github.com/hashicorp/terraform-provider-google/blob/f2f60b641c1dac08f38c9ebf9f826dca63c7d403/google/services/container/resource_container_cluster.go#L2546-L2565

Reply:
I'm assuming nodepool comes from the Marvel 1852142180 universe. ;)